src/plugins/guest-comment-mixin.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * This mixin is used to submit a comment on a loan that
+ * was left as a guest before they created their account.
+ * If the query parameter "guest_comment" is present and
+ * the cookie values are present, and the user is logged in,
+ * we can now leave the comment.
+ */
+import { gql } from '@apollo/client';
+import logFormatter from '@/util/logFormatter';
+
+export default {
+	mounted() {
+		const guestCommentComment = this.cookieStore.get('guestCommentComment');
+		const guestCommentInProgress = this.cookieStore.get('guestCommentInProgress');
+		const guestCommentLoanId = this.cookieStore.get('guestCommentLoanId');
+		const currentLoanId = this.$route?.params?.id;
+		const isTargetBorrowerProfile = currentLoanId.toString() === guestCommentLoanId.toString();
+		if (this.$route.query.guest_comment
+			&& isTargetBorrowerProfile
+			&& guestCommentComment
+			&& guestCommentInProgress) {
+			this.submitComment();
+		}
+	},
+	methods: {
+		submitComment() {
+			this.apollo.mutate({
+				mutation: gql`mutation commentOnLoan($id: Int!, $body: String) {
+					loan(id: $id) {
+						addComment(body: $body)
+					}
+				}`,
+				variables: {
+					id: Number(this.$route?.params?.id ?? 0),
+					body: this.cookieStore.get('guestCommentComment')
+				}
+			}).then(({ data }) => {
+				// comment was added successfully
+				if (data.loan.addComment) {
+					this.$showTipMsg('Thank you for leaving a comment!');
+					// remove cookie values so we don't post the comment again
+					this.cookieStore.remove('guestCommentComment');
+					this.cookieStore.remove('guestCommentInProgress');
+					this.cookieStore.remove('guestCommentLoanId');
+				} else {
+					throw new Error('Comment not added');
```

Comment:
Should we remove the cookie also if it was unsuccessful?

Reply:
Maybe? My thinking was that if there was some intermittent server error or something the comment would still be added if the user reloaded the page or visited the loan next time.